src/resumable-upload.ts

```diff
@@ -356,10 +365,34 @@ export class Upload extends Writable {
     ];
     this.authClient = cfg.authClient || new GoogleAuth(cfg.authConfig);
 
-    this.apiEndpoint = 'https://storage.googleapis.com';
-    if (cfg.apiEndpoint) {
+    const universe = cfg.universeDomain || DEFAULT_UNIVERSE;
+
+    this.apiEndpoint = `https://storage.${universe}`;
+    if (cfg.apiEndpoint && cfg.apiEndpoint !== this.apiEndpoint) {
       this.apiEndpoint = this.sanitizeEndpoint(cfg.apiEndpoint);
-      if (!DEFAULT_API_ENDPOINT_REGEX.test(cfg.apiEndpoint)) {
+
+      const hostname = new URL(this.apiEndpoint).hostname;
+
+      // check if it is a domain of a known universe
+      const isDomain = hostname === universe;
+      const isDefaultUniverseDomain = hostname === DEFAULT_UNIVERSE;
+
+      // check if it is a subdomain of a known universe
+      // by checking a last (universe's length + 1) of a hostname
+      const isSubDomainOfUniverse =
+        hostname.slice(-(universe.length + 1)) === `.${universe}`;
+      const isSubDomainOfDefaultUniverse =
+        hostname.slice(-(DEFAULT_UNIVERSE.length + 1)) ===
+        `.${DEFAULT_UNIVERSE}`;
+
+      if (
+        !isDomain &&
+        !isDefaultUniverseDomain &&
+        !isSubDomainOfUniverse &&
+        !isSubDomainOfDefaultUniverse
+      ) {
+        // a custom, non-universe domain,
+        // use gaxios
```

Comment:
Going to create a follow-up issue to clean this up - we should have a cleaner, explicit mechanism for disabling/bypassing auth; even for emulator contexts.

Reply:
Created:
- https://github.com/googleapis/nodejs-storage/issues/2400